src/Domain/WsServer/WsServer.php

```diff
@@ -213,413 +189,152 @@ public function onError(ConnectionInterface $conn, Exception $e)
-                        ]);
-                        $this->clients[$connResourceId]->send($json);
-                        return $batchResultContainer;
-                    },
-                    function ($rejection) use ($connResourceId) {
-                        $reason = $rejection;
-                        if ($rejection instanceof ExecuteBatchRejection) {
-                            $batchId = $rejection->getBatchId();
-                            $reason = $rejection->getReason();
-                            $message = '';
-                            if (is_string($reason)) {
-                                $message = $reason;
-                            }
-                            while ($reason instanceof Throwable) {
-                                $message .= $reason->getMessage() . PHP_EOL;
-                                $reason = $reason->getPrevious();
-                            }
-                            $json = json_encode([
-                                'header_type' => 'Batch/ExecuteBatch',
-                                'header_data' => [
-                                    'batch_id' => $batchId,
-                                ],
-                                'success' => false,
-                                'message' => $message ?: 'Unknown reason',
-                                'payload' => null
-                            ]);
-                            $this->clients[$connResourceId]->send($json);
-                            return []; // do not propagate rejection, just resolve to empty batch results
-                        }
-                        return reject($reason);
-                    }
-                );
-            }
-        }
-        return all($promises);
     }
 
-    private function comparePayloads(array $p1, array $p2): EPayloadDifferenceType
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function getGameSessionIds(): PromiseInterface
     {
-        // any state change is essential
-        if ($p1['tick']['state'] !== $p2['tick']['state']) {
-            return new EPayloadDifferenceType(EPayloadDifferenceType::ESSENTIAL_DIFFERENCES);
-        }
-
-        // remember for later
-        $p1TickEraTimeleft = $p1['tick']['era_timeleft'] ?? 0;
-        $p2TickEraTimeleft = $p2['tick']['era_timeleft'] ?? 0;
-        $eraTimeLeftDiff = abs($p1TickEraTimeleft - $p2TickEraTimeleft);
-
-        // if there are any other changes then "time" fields, it is essential
-        unset(
-            $p1['prev_update_time'],
-            $p1['update_time'],
-            $p1TickEraTimeleft,
-            $p1['tick']['era_timeleft'],
-            $p2['prev_update_time'],
-            $p2['update_time'],
-            $p2TickEraTimeleft,
-            $p2['tick']['era_timeleft']
+        $connection = AsyncDatabase::createServerManagerConnection(Loop::get());
+        $qb = $connection->createQueryBuilder();
+        return $connection->query(
```

Comment:
let's check if the game_list record is created after all the session data has been created list the msp_session_x database itself